include/deal.II/base/bounding_box.h

```diff
@@ -77,14 +110,22 @@ class BoundingBox
   const std::pair<Point<spacedim,Number>,Point<spacedim,Number>> &get_boundary_points () const;
 
   /**
+   * Check if the current object and @p other_bbox are neighbors, i.e. if the boxes
+   * have dimension spacedim, check if their intersection is non empty
+   *
+   * Return an enumerator of type NeighborType.
+   */
+  NeighborType is_neighbor(const BoundingBox<spacedim,Number> &other_bbox) const;
```

Comment:
I must admit I also prefer NOT to call it `is_neighbor` is this implies `bool`. Maybe something like `get_neihbor_type()`

Reply:
Ok, get_neighbor_kind I like ;)

Reply:
If the return type is called `NeighborType`, then the function should be called `get_neighbor_type()`, not `get_neighbor_kind()`. But you can of course also rename the type :-)

Reply:
>  then the function should be called get_neighbor_type(), not get_neighbor_kind().

exactly what i suggested above :smile: